File: exercise/echo-function/evaluation/evaluator.py
from evaluation_utils import EvaluationResult


def evaluate(actual):
    correct = actual == "correct"
    return EvaluationResult(correct, "correct", actual, ["Hallo"])


def evaluate_value(expected, actual, messages):
    return EvaluationResult(expected == actual, expected, actual, ["Hallo"])

File: exercise/echo-function/evaluation/Evaluator.hs
{-# LANGUAGE ScopedTypeVariables #-}
module Evaluator where

import EvaluationUtils
import Control.Exception

evaluate :: String -> EvaluationResult
evaluate value  =
    let correct = if value == "correct" then True else False
    in evaluationResult {
        result = correct,
        readableExpected = Just "correct",
        readableActual = Just value,
        messages = ["Hallo"]
    }


evaluate_value :: String -> String -> [String] -> EvaluationResult
evaluate_value expected actual arguments =
    let correct = if actual == expected then True else False
    in evaluationResult {
        result = correct,
        readableExpected = Just expected,
        readableActual = Just actual,
        messages = ["Hallo"]
    }